src/jailer/src/env.rs

```diff
@@ -674,16 +674,45 @@ impl Env {
 
         // Daemonize before exec, if so required (when the dev_null variable != None).
         if let Some(dev_null) = dev_null {
-            // Call setsid().
+            // We follow the double fork method to daemonize the jailer referring to
+            // https://0xjet.github.io/3OHA/2022/04/11/post.html
+            // setsid() will fail if the calling process is a process group leader.
+            // By calling fork(), we guarantee that the newly created process inherits
+            // the PGID from its parent and, therefore, is not a process leader.
             // SAFETY: Safe because it's a library function.
-            SyscallReturnCode(unsafe { libc::setsid() })
-                .into_empty_result()
-                .map_err(JailerError::SetSid)?;
-
-            // Replace the stdio file descriptors with the /dev/null fd.
-            dup2(dev_null.as_raw_fd(), STDIN_FILENO)?;
-            dup2(dev_null.as_raw_fd(), STDOUT_FILENO)?;
-            dup2(dev_null.as_raw_fd(), STDERR_FILENO)?;
+            let child_pid = unsafe { libc::fork() };
+            if child_pid < 0 {
+                return Err(JailerError::Daemonize("Failed to fork".to_string()));
+            } else if child_pid != 0 {
+                // parent exiting
+                exit(0);
+            } else {
```

Comment:
LGTM.

style nit: since we exit early in the parent, would it make sense to change to avoid the if nesting?

```rust
if child_pid != 0 {
    exit(0);
}
// no need for else
```